dashboard/lib/services/curriculum_pdfs.rb

```diff
@@ -0,0 +1,87 @@
+require 'cdo/chat_client'
+require 'dynamic_config/dcdo'
+
+module Services
+  # Contains all code related to the generation, storage, and
+  # retrieval of Curriculum PDFs.
+  #
+  # Also see curriculum_pdfs.rake for some associated logic
+  module CurriculumPdfs
+    include LessonPlans
+    include Utils
+
+    DEBUG = false
+    S3_BUCKET = "cdo-lesson-plans#{'-dev' if DEBUG}".freeze
+
+    # Whether or not we should generate PDFs. Specifically, this
+    # encapsulates three concerns:
+    #
+    # 1. Is this code running on the staging server? We only want to do this
+    #    as part of the staging build; the generated PDFs will be made
+    #    available to other environments, so they don't need to run this
+    #    process themselves.
+    # 2. Is the script one for which we care about PDFs? Right now, we only
+    #    want to generate PDFs for "migrated" scripts.
+    # 3. Is the script actually being updated? The overall seed process is
+    #    indiscriminate, and will happily re-seed content even without
+    #    changes. This is fine for database upserts, but we want to be more
+    #    cautious with the more-expensive PDFs generation process.
+    #
+    # In addition, we support manually disabling this feature with DCDO
+    def self.generate_pdfs?(script_data)
+      return true if DEBUG
+
+      return false unless rack_env?(:staging)
```

Comment:
this might want to become a CDO variable at some point, to make it easier to test this feature locally. I'm also not sure what rack_env staging-next runs as, but if it runs as `:staging` then that might be another reason to do this.